### DISCUSSION TITLE
Is there a Gleam book or even a PDF/EPUB of docs?

### DISCUSSION THREAD

**Discussion by impredicative** [gleam-lang/gleam#2622] [Ideas & suggestions]
I'd like a single Gleam book that I can read through. If nothing else, it can be made from the docs. It can be made available as PDF and EPUB. I would prefer it over reading scattered individual docs.

**Comment by lpil** [comment]
No one has made one that I'm aware of.

**Comment by han-tyumi** [comment]
I'm also one to prefer a more book-like experience.
Rust's is one that comes to mind that I've liked in the past.
https://doc.rust-lang.org/book/

However, I've been finding the [language tour](https://tour.gleam.run/) pretty useful when needing to reference some feature quickly.
It just feels a bit clunky/heavy to navigate over something like the Rust book above.

**Reply by lpil** [reply]
We would like a book but unfortunately we don't have any funding to make one.

**Reply by impredicative** [reply]
@lpil Can something tentatively just be compiled from preexisting docs that would go well together in the right sequence?

**Reply by lpil** [reply]
Would all the lessons of the tour on one page suffice?

**Reply by impredicative** [reply]
@lpil Yes, it should be a useful start for now. In it, each lesson's corresponding code and output should follow its respective text.

**Reply by lpil** [reply]
Fab. I've made an issue here: https://github.com/gleam-lang/language-tour/issues/36

**Reply by impredicative** [reply]
The language tour is now available in a single page at https://tour.gleam.run/everything/ but it's still not possible to print it to pdf, etc.

**Reply by lpil** [reply]
I've made an issue for that here: <https://github.com/gleam-lang/language-tour/issues/71>

**Reply by impredicative** [reply]
The language tour /everything/ page can now be printed to pdf, at least via Firefox.

**Comment by impredicative** [comment]
@blksnk To confirm, is the /everything/ page updated automatically when:

1. any individual tour page is updated
2. a new tour page is added

**Reply by lpil** [reply]
Yup it's all generated from the same source files

**Reply by blksnk** [reply]
Exactly
